examples/seismic/skew_self_adjoint/ssa_01_iso_implementation.ipynb

```diff
@@ -0,0 +1,1452 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "jupyter": {
+     "source_hidden": true
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from examples.seismic import RickerSource, Receiver, TimeAxis\n",
+    "from devito import (Grid, Function, TimeFunction, SpaceDimension, Constant, \n",
+    "                    Eq, Operator, solve, configuration)\n",
+    "from devito.finite_differences import Derivative\n",
+    "from devito.builtins import gaussian_smooth\n",
+    "import matplotlib as mpl\n",
+    "import matplotlib.pyplot as plt\n",
+    "from matplotlib import cm\n",
+    "from timeit import default_timer as timer\n",
+    "%matplotlib inline\n",
+    "mpl.rc('font', size=14)\n",
+    "\n",
+    "# We also define 32 bit floating point as the precision type \n",
+    "dtype = np.float32\n",
+    "\n",
+    "# set the default language to openmp\n",
+    "configuration['language'] = 'openmp'\n",
+    "\n",
+    "# set the logging to debug\n",
+    "configuration['log-level'] = 'DEBUG'"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Unit test demonstrating skew symmetry for shifted derivatives\n",
+    "\n",
+    "As noted above, we will demonstrate with a small 1D unit test and 8th order spatial operator that the devito shifted first derivatives are skew symmetric. This anti-symmetry can demonstrated for the forward and backward half cell shift first derivative operators $\\overrightarrow{\\partial}$ and $\\overleftarrow{\\partial}$ with two random vectors $x_1$ and $x_2$ by verifying the following *dot product test* equation (2) written above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "Allocating memory for f1(117,)\n",
+      "Allocating memory for g1(117,)\n",
+      "Operator `Kernel` generated in 0.15 s\n",
+      "  * lowering.Expressions: 0.09 s (61.4 %)\n",
+      "  * lowering.IET: 0.03 s (20.5 %)\n",
+      "     * specializing.IET: 0.03 s (20.5 %)\n",
+      "  * lowering.Clusters: 0.03 s (20.5 %)\n",
+      "     * specializing.Clusters: 0.03 s (20.5 %)\n",
+      "Flops reduction after symbolic optimization: [62 --> 25]\n",
+      "Allocating memory for f2(117,)\n",
+      "Allocating memory for g2(117,)\n",
+      "Operator `Kernel` fetched `/tmp/devito-jitcache-uid5138/8a11eb4be1534b4bdb8640c8c4100d243ec05e3f.c` in 0.12 s from jit-cache\n",
+      "Operator `Kernel` run in 0.01 s\n",
+      "* section0<101> with OI=1.51 computed in 0.01 s [0.01 GFlops/s]\n",
+      "Performance[mode=advanced] arguments: {'nthreads': 16}\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "f1g2,g1f2,diff; -3.40460658e-01 -3.40461075e-01 -6.12746646e-07\n"
```

Comment:
f1g2, g1f2, diff

Reply:
are you OK with full colon ":"? is ";" stylistically bad?

Reply:
My comment was about space after comma, about semicolon I am fine.

Reply:
got it